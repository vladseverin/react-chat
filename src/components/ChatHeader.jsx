import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid'
import Avatar from './Avatar.jsx';
import UserMenu from './UserMenu.jsx';
import ChatMenu from './ChatMenu.jsx';

const styles = theme => ({
  appBar: {
    width: 'calc(100% - 320px)',
    position: 'fixed',
  },
  appBarTitle: {
    marginLeft: theme.spacing.unit * 2,
    color: theme.palette.secondary.contrastText,
    whiteSpace: 'nowrap',
  },
});

class ChatHeader extends React.Component {
  render() {
    const { 
      classes, logout, activeUser, activeChat, leaveChat, deleteChat, editUser, isConnected, soketsDisconnect
    } = this.props; 

    return (
      <AppBar className={classes.appBar} >
        <Toolbar>
          <Grid container spacing={24} direction='row' justify='space-between' alignItems='center' wrap='nowrap'>
            <Grid container direction='row' wrap='nowrap' justify='flex-start' alignItems='center'>
              {activeChat ? (
                <React.Fragment>
                  <Avatar colorFrom={activeChat._id}>
                    {activeChat.title}
                  </Avatar>
                  <Typography variant="title" className={classes.appBarTitle}>
                    {activeChat.title}
                  </Typography>
                  <ChatMenu
                    disabled={!isConnected}
                    activeUser={activeUser}
                    onLeaveClick={() => leaveChat(activeChat._id)}
                    onDeleteClick={() => deleteChat(activeChat._id)}
                  />
                </React.Fragment>
              ) : (
              <Typography variant="title" color="inherit" noWrap>
                Juicy Chat
              </Typography>
              )}

            </Grid>
            <Grid item> 
              <UserMenu
                disabled={!isConnected}
                soketsDisconnect={soketsDisconnect}
                activeUser={activeUser}
                logout={logout}
                editUser={editUser}
              />
            </Grid>
          </Grid>
        </Toolbar>
      </AppBar>
    );
  }
}
export default withStyles(styles)(ChatHeader);
