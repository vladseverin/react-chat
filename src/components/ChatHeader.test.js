import React from 'react';
import ReactDOM from 'react-dom';
import ChatHeader from './ChatHeader';

jest.mock('./Avatar', () => () => 'Avatar');
jest.mock('./UserMenu', () => () => 'UserMenu');
jest.mock('./ChatMenu', () => () => 'ChatMenu');

const mockProps = {
  activeUser: {
    firstName: 'Name',
    lastName: 'Surname',
    username: 'username',
    isMember: true,
    isCreator: true,
    isChatMember: true,
  },
  activeChat: {
    createdAt: '2018-06-13T20:48:03.200Z',
    creator: {
      firstName: 'firstName',
      lastName: 'lastName',
      username: 'username',
      _id: '12345',
    },
    members: [
      {
        firstName: 'firstName',
        lastName: 'lastName',
        username: 'username',
        _id: '12345',
      },
    ],
    _id: '12345',
    title: 'myChat',
  },
  logout: jest.fn(),
  leaveChat: jest.fn(),
  deleteChat: jest.fn(),
  editUser: jest.fn(),
  isConnected: true,
};

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ChatHeader {...mockProps} />, div);
  ReactDOM.unmountComponentAtNode(div);
});
