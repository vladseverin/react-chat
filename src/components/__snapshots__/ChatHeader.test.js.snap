// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatHeader /> renders correctly 1`] = `
<header
  className="MuiPaper-root-165 MuiPaper-elevation4-171 MuiAppBar-root-157 MuiAppBar-positionFixed-158 MuiAppBar-colorPrimary-163 mui-fixed ChatHeader-appBar-155"
>
  <div
    className="MuiToolbar-root-192 MuiToolbar-gutters-193"
  >
    <div
      className="MuiGrid-container-194 MuiGrid-spacing-xs-24-217 MuiGrid-wrap-xs-nowrap-200 MuiGrid-align-items-xs-center-202 MuiGrid-justify-xs-space-between-213"
    >
      <div
        className="MuiGrid-container-194 MuiGrid-wrap-xs-nowrap-200 MuiGrid-align-items-xs-center-202"
      >
        Avatar
        <h2
          className="MuiTypography-root-285 MuiTypography-title-291 ChatHeader-appBarTitle-156"
        >
          myChat
        </h2>
        ChatMenu
      </div>
      <div
        className="MuiGrid-item-195"
      >
        UserMenu
      </div>
    </div>
  </div>
</header>
`;
